### PR TITLE
Add hasParameters and hasVariables functions

### LINKED ISSUE #28: Add hasParameters to Channel object
It would be great to have a convenient method for the Channel object to check if it has parameters defined. Something like:

```js
channel.hasParameters()
```

### PR DESCRIPTION
**Pull request by derberg**
Shame but I've never used double bang and had to google. I love it. You learn your entire life.

Fixes #28

### CODE REVIEW

File: test/models/asyncapi_test.js [hunk 1]
```diff
@@ -73,6 +73,17 @@ describe('AsyncAPIDocument', () => {
     });
   });
 
+  describe('#hasVariable()', function () {
```
Comment (fmvilas) [inline test/models/asyncapi_test.js]:
```suggestion
  describe('#hasVariables()', function () {
```

File: test/models/asyncapi_test.js [hunk 2]
```diff
@@ -73,6 +73,17 @@ describe('AsyncAPIDocument', () => {
...
+    it('should return a boolean indicating if the AsyncAPI document has server variables', () => {
```
Comment (fmvilas) [inline test/models/asyncapi_test.js]:
```suggestion
    it('should return a boolean indicating if a server URL has variables', () => {
```

File: test/models/asyncapi_test.js [hunk 3]
```diff
@@ -73,6 +73,17 @@ describe('AsyncAPIDocument', () => {
...
+      const doc = { servers: { test1: { url: 'test1', variables: { port: { desc: 'test1' } } } } };
```
Comment (fmvilas) [inline test/models/asyncapi_test.js]:
```suggestion
      const doc = { servers: { test1: { url: 'test1:{port}', variables: { port: { desc: 'test1' } } } } };
```

I know this is not strictly needed but would be good for clarity.

File: test/models/asyncapi_test.js [hunk 4]
```diff
@@ -73,6 +73,17 @@ describe('AsyncAPIDocument', () => {
...
+      const docNoServerVariables = { servers: { test: { url: 'test' } } };
+      const d = new AsyncAPIDocument(doc);
+      const d2 = new AsyncAPIDocument(docNoServerVariables);
+      expect(d.server('test1').hasVariables()).to.equal(true);
+      expect(d2.server('test').hasVariables()).to.equal(false);
+    });
+  });
```
Comment (fmvilas) [inline test/models/asyncapi_test.js]: I think this test should be placed here: https://github.com/asyncapi/parser-js/blob/master/test/models/server_test.js

File: test/models/asyncapi_test.js [hunk 5]
```diff
@@ -126,6 +137,17 @@ describe('AsyncAPIDocument', () => {
...
+    it('should return a boolean indicating if the AsyncAPI document has channel parameters', () => {
+      const doc = { channels: { test1: { parameters: { test1param: { description: "test1param" } } } } };
+      const docNoChannelParams = { channels: { test: { description: "test" } } };
+      const d = new AsyncAPIDocument(doc);
+      const d2 = new AsyncAPIDocument(docNoChannelParams);
+      expect(d.channel('test1').hasParameters()).to.equal(true);
+      expect(d2.channel('test').hasParameters()).to.equal(false);
+    });
+  });
+
```
Comment (fmvilas) [inline test/models/asyncapi_test.js]: I think this test should be here: https://github.com/asyncapi/parser-js/blob/master/test/models/channel_test.js

File: test/models/asyncapi_test.js (hunk 4 above)
Comment (derberg) [reply test/models/asyncapi_test.js]: damn, right 🤦‍♂ not my day

**Comment by derberg** [comment]
@jonaslagoni you might wanna have a look at this one because of https://github.com/asyncapi/generator/pull/156

**Pull request merged**